cmd/convert.go

```
@@ -6,117 +6,79 @@ package cmd
 import (
 	"fmt"
 	"strconv"
-	"strings"
 
 	"github.com/Achno/gowall/internal/image"
+	imageio "github.com/Achno/gowall/internal/image_io"
+	"github.com/Achno/gowall/internal/logger"
 	"github.com/Achno/gowall/utils"
 	"github.com/spf13/cobra"
 )
 
-var formatFlag string
-var colorPair []string
-var outputName string
+var (
+	colorPair []string
+	theme     string
+)
 
 var convertCmd = &cobra.Command{
-	Use:   "convert [image path / batch flag]",
+	Use:   "convert [INPUT] [OPTIONAL OUTPUT]",
 	Short: "Convert an img's color scheme",
-	Long:  `Convert an img's color scheme`,
-	// Args: cobra.MinimumNArgs(1),
+	Long:  `Convert an img's color scheme or its format ie from webp to png etc`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		err := validateInput(shared, args)
+		if err != nil {
```

This should use `utils.HandleError()` instead of returning err, you dont have anywhere else to bubble this error. Also introduces inconsistencies because the Error will not be printed with red text but white.